File: doan/lib/models/booking.dart
class Booking {
  final int? id;
  final int userId;
  final int showtimeId;
  final String bookingDate;
  final double totalPrice;
  final String paymentMethod;
  final String paymentStatus;

  Booking({
    this.id,
    required this.userId,
    required this.showtimeId,
    required this.bookingDate,
    required this.totalPrice,
    required this.paymentMethod,
    required this.paymentStatus,
  });

  factory Booking.fromMap(Map<String, dynamic> map) => Booking(
    id: map['ID'],
    userId: map['USERID'],
    showtimeId: map['SHOWTIMEID'],
    bookingDate: map['BOOKINGDATE'],
    totalPrice: map['TOTALPRICE'].toDouble(),
    paymentMethod: map['PAYMENTMETHOD'],
    paymentStatus: map['PAYMENTSTATUS'],
  );

  Map<String, dynamic> toMap() => {
    'ID': id,
    'USERID': userId,
    'SHOWTIMEID': showtimeId,
    'BOOKINGDATE': bookingDate,
    'TOTALPRICE': totalPrice,
    'PAYMENTMETHOD': paymentMethod,
    'PAYMENTSTATUS': paymentStatus,
  };
}

File: backend/bookingmovieticketapp/src/main/java/com/project/bookingmovieticketapp/Models/Booking.java
package com.project.bookingmovieticketapp.Models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "bookings")
@Builder
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name ="userid")
    private User user;

    @ManyToOne
    @JoinColumn(name = "showtimeid")
    private ShowTime showTime;

    @Column(name = "bookingdate", nullable = false)
    private LocalDateTime bookingdate;

    @Column(name = "totalprice",nullable = false)
    private int totalprice;

    @Column(name = "paymentmethod",nullable = false)
    private int paymentmethod;

    @Column(name = "paymentstatus",nullable = false)
    private int paymentstatus;
}
